include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -1883,4 +1892,154 @@
   }
 }
 
+inline ShapeInferContext::ShapeInferContext(const OrtApi* ort_api,
+                                            OrtShapeInferContext* ctx) : ort_api_(ort_api), ctx_(ctx) {
+  size_t input_count = 0;
+  Ort::ThrowOnError(ort_api_->ShapeInferContext_GetInputCount(ctx_, &input_count));
+  for (size_t ith_input = 0; ith_input < input_count; ++ith_input) {
+    OrtTensorTypeAndShapeInfo* info{};
+    Ort::ThrowOnError(ort_api_->ShapeInferContext_GetInputTypeShape(ctx, ith_input, &info));
+    TensorTypeAndShapeInfo type_shape_info(info);
+    auto integer_shape = type_shape_info.GetShape();
+    std::vector<const char*> symbolic_shape(integer_shape.size(), {});
+    type_shape_info.GetSymbolicDimensions(&symbolic_shape[0], integer_shape.size());
+    Shape shape;
+    for (size_t ith = 0; ith < integer_shape.size(); ++ith) {
+      if (integer_shape[ith] == -1) {
```

Comment:
>if (integer_shape[ith] == -1)  [](http://example.com/codeflow?start=6&length=30)

i remember onnx use "-1" as special shape in op like Reshape. use -1 to check seem not safe.

Reply:
Removed.